src/mpid/ch4/netmod/ofi/ofi_events.h

```diff
@@ -130,8 +130,18 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_recv_event(struct fi_cq_tagged_entry *wc,
         }
     }
     else if (MPIDI_OFI_ENABLE_PT2PT_NOPACK && (event_id == MPIDI_OFI_EVENT_RECV_NOPACK) && (MPIDI_OFI_REQUEST(rreq, noncontig.nopack))) {
-            last = count;
-            MPL_free(MPIDI_OFI_REQUEST(rreq, noncontig.nopack));
+        MPI_Count elements;
+
+        /* Check to see if there are any bytes that don't fit into the datatype basic elements */
+        MPIR_Get_elements_x_impl(((MPI_Count *) &count), MPIDI_OFI_REQUEST(rreq, datatype), &elements);
+        if (count) {
```

Comment:
Could we just return `MPI_ERR_TYPE` from the get elements function instead?

Reply:
I don't think it's an error in that case.

Reply:
I'm just looking at the way this is used, and it doesn't do anything with the count information besides check if it's non-zero. Isn't that what returning `MPI_UNDEFINED` in the elements argument already indicates?

Reply:
Does it always return `MPI_UNDEFINED` in that case? I'm not sure that's true. If so, then we could check that instead of a count > 1, but I'm not sure if we should return an error from `MPIR_Get_elements_x_impl` or not. I don't know if there's a valid use case.

Reply:
You are right that it is not consistent in returning undefined. Seems like a bug to me. Going to keep digging...